src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) throws SalaryInfoException {
+        if (names.length == 0 || data.length == 0 || dateFrom == null || dateTo == null) {
+            throw new SalaryInfoException("incorrect data entry");
+        }
+        DateTimeFormatter df = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateFrom1 = LocalDate.parse(dateFrom.trim(), df);
+        LocalDate dateTo1 = LocalDate.parse(dateTo.trim(), df);
+        String info;
+        StringBuilder stringBuilder = new StringBuilder();
+        int[]sums = new int[names.length];
+        for (int i = 0; i < names.length; i++) {
+            for (String dat : data) {
+                String[] dateLine = dat.split(" ");
+                LocalDate d1 = LocalDate.parse(dateLine[0].trim(), df);
+                if (names[i].equals(dateLine[1]) && (d1.isAfter(dateFrom1.minusDays(1))
+                        && d1.isBefore(dateTo1.plusDays(1)))) {
```

Comment:
extract isAfter and isBefore to separate method, to clarify the conditions by proper method name